lambdas/data-migration2/tests/test-granulesAndFiles.js

```diff
@@ -351,20 +351,103 @@ test.serial('migrateGranuleRecord handles nullable fields on source granule data
   );
 });
 
-test.serial('migrateGranuleRecord throws RecordAlreadyMigrated error for already migrated record', async (t) => {
+test.serial('migrateGranuleRecord throws RecordAlreadyMigrated error if previously migrated record is newer', async (t) => {
   const {
     knex,
     testGranule,
   } = t.context;
 
-  await migrateGranuleRecord(testGranule, knex);
+  const testGranule1 = testGranule;
+  const testGranule2 = {
+    ...testGranule1,
+    updatedAt: Date.now() - 1000,
+  };
+
+  await migrateGranuleRecord(testGranule1, knex);
 
   await t.throwsAsync(
-    migrateGranuleRecord(testGranule, knex),
+    migrateGranuleRecord(testGranule2, knex),
+    { instanceOf: RecordAlreadyMigrated }
+  );
+});
+
+test.serial('migrateGranuleRecord throws RecordAlreadyMigrated error the record was previously migrated and the new record is "running"', async (t) => {
+  const {
+    knex,
+    testGranule,
+  } = t.context;
+
+  const testGranule1 = testGranule;
+  const testGranule2 = {
+    ...testGranule1,
+    status: 'running',
+    updatedAt: Date.now(),
+  };
+
+  await migrateGranuleRecord(testGranule1, knex);
+
+  await t.throwsAsync(
+    migrateGranuleRecord(testGranule2, knex),
     { instanceOf: RecordAlreadyMigrated }
   );
 });
 
+test.serial('migrateGranuleRecord updates an already migrated record if the updated date is newer', async (t) => {
+  const {
+    knex,
+    granulePgModel,
+    testCollection,
+    testExecution,
+    collectionCumulusId,
+  } = t.context;
+
+  const testGranule = generateTestGranule({
+    collectionId: buildCollectionId(testCollection.name, testCollection.version),
+    execution: testExecution.url,
+    status: 'completed',
+  });
+
+  await migrateGranuleRecord(testGranule, knex);
+
+  const newerGranule = {
+    ...testGranule,
+    updatedAt: Date.now(),
+  };
+
+  const [granuleCumulusId] = await migrateGranuleRecord(newerGranule, knex);
+  const record = await granulePgModel.get(knex, {
+    cumulus_id: granuleCumulusId,
+  });
+
+  t.deepEqual(
```

Comment:
I don't know if we need to do a deepEqual comparison to the whole record. We only need to prove that the properties that should have been updated were changed to the correct value, so in this case the `updated_at` timestamp

Reply:
https://github.com/nasa/cumulus/pull/2162/commits/171504648b8a39c26376eaa6d4931da6b181c2e2